packages/ckeditor5-dev-tests/lib/utils/manual-tests/compilemanualtesthtmlfiles.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* jshint node: true, strict: true */
+
+'use strict';
+
+const path = require( 'path' );
+const { logger } = require( '@ckeditor/ckeditor5-dev-utils' );
+const globSync = require( '../glob' );
+const fs = require( 'fs-extra' );
+const gutil = require( 'gulp-util' );
+const commonmark = require( 'commonmark' );
+const combine = require( 'dom-combiner' );
+const getRelativeFilePath = require( '../getrelativefilepath' );
+
+const reader = new commonmark.Parser();
+const writer = new commonmark.HtmlRenderer();
+
+const viewTemplate = fs.readFileSync( path.join( __dirname, 'template.html' ), 'utf-8' );
+
+module.exports = function compileManualTestHtmlFiles( buildDir, manualTestPattern ) {
+	const sourceMDFiles = globSync( path.join( manualTestPattern, '*.md' ) );
+	const sourceDirs = uniq( sourceMDFiles.map( file => path.dirname( file ) ) );
+
+	fs.ensureDirSync( buildDir );
+
+	// Copy all files which can be found in the directories with manual tests
+	// to the build dir.
+	sourceDirs.forEach( sourceDir => copyStaticFiles( buildDir, sourceDir ) );
+
+	// Generate real HTML files out of the MD + HTML files of each test.
+	sourceMDFiles.forEach( sourceFile => compileTestHtmlFile( buildDir, sourceFile ) );
+};
+
+function compileTestHtmlFile( buildDir, sourceFile ) {
+	const log = logger();
+
+	const relativeFilePath = getRelativeFilePath( sourceFile );
+
+	log.info( `Processing '${ gutil.colors.cyan( sourceFile ) }'...` );
+
+	// Compile test instruction (Markdown file).
+	const parsedMarkdownTree = reader.parse( fs.readFileSync( sourceFile, 'utf-8' ) );
+	const manualTestInstructions = '<div class="manual-test-sidebar">' + writer.render( parsedMarkdownTree ) + '</div>';
+
+	// Load test view (HTML file).
+	const htmlView = fs.readFileSync( `${ changeExtension( sourceFile, 'html' ) }`, 'utf-8' );
+
+	// Attach script file to the view.
+	const scriptTag =
+		'<body class="manual-test-container">' +
+			`<script src="./${ changeExtension( path.basename( sourceFile ), 'js' ) }"></script>` +
+		'</body>';
+
+	// Concat the all HTML parts to single one.
+	const preparedHtml = combine( viewTemplate, manualTestInstructions, htmlView, scriptTag );
+
+	// Prepare output path.
+	const outputFilePath = path.join( buildDir, changeExtension( relativeFilePath, 'html' ) );
+
+	fs.outputFileSync( outputFilePath, preparedHtml );
+
+	log.info( `Finished writing '${ gutil.colors.cyan( outputFilePath ) }'` );
+}
+
+// Copies all non JS/HTML/MD files to build dir. Their relative paths to JS/HTML files are maintained.
+function copyStaticFiles( buildDir, sourceDir ) {
+	const files = globSync( path.join( sourceDir, '**', '*.!(js|html|md)' ) );
```

Comment:
If some manual test will require another HTML file (for example source for `iframe`), it won't work as we expected because the source file won't be copied.

[That's the reason why we had a little more complicated the method.](https://github.com/ckeditor/ckeditor5-dev-tests/blob/master/lib/utils.js#L362-L401)